discovery/db_discovery.go

```diff
@@ -0,0 +1,92 @@
+package discovery
+
+import (
+	"time"
+
+	"github.com/livepeer/go-livepeer/common"
+	"github.com/livepeer/go-livepeer/core"
+	lpTypes "github.com/livepeer/go-livepeer/eth/types"
+	"github.com/livepeer/go-livepeer/net"
+
+	"github.com/golang/glog"
+)
+
+type DBOrchestratorPoolCache struct {
+	node *core.LivepeerNode
+}
+
+func NewDBOrchestratorPoolCache(node *core.LivepeerNode) *DBOrchestratorPoolCache {
+	if node.Eth == nil {
+		glog.Error("Could not refresh DB list of orchestrators: LivepeerNode nil")
+		return nil
+	}
+
+	ticker := time.NewTicker(1 * time.Hour)
+	go func(node *core.LivepeerNode) {
+		for {
+			select {
+			case <-ticker.C:
+				orchestrators, err := node.Eth.RegisteredTranscoders()
+				if err != nil {
+					glog.Error("Could not refresh DB list of orchestrators: ", err)
+					return
+				}
+				_, dbOrchErr := CacheDBOrchs(node, orchestrators)
+				if dbOrchErr != nil {
+					glog.Error("Could not refresh DB list of orchestrators: CacheDBOrchs err")
+					return
+				}
+			}
+		}
+	}(node)
+
+	return &DBOrchestratorPoolCache{node: node}
+}
+
+func CacheDBOrchs(node *core.LivepeerNode, orchs []*lpTypes.Transcoder) ([]*common.DBOrch, error) {
+	var dbOrchs []*common.DBOrch
+	if orchs == nil {
+		glog.Error("No new DB orchestrators created: no orchestrators found onchain")
+		return dbOrchs, nil
+	}
+
+	for _, orch := range orchs {
+		dbOrch := EthOrchToDBOrch(orch)
+		if err := node.Database.UpdateOrchs(dbOrch); err != nil {
```

Comment:
Just a note for later. At some point we'll probably want to do one of a few things (or even a combination of both):

1. Call `GetOrchestratorInfo` prior to inserting to DB to receive and insert ticket parameters. We need to figure out how to schedule this for large O sets (sequentially? asynchronously? in batches?)
2. Consolidate multiple INSERTs into a single statement, eg `INSERT INTO orchestrators (updatedAt, serviceURI, ethereumAddr) VALUES(datetime(), abc, def), (datetime(), ghi, jkl), (datetime(), mno, pqr)`. We want to minimize the amount of time SQLite spends writing to disk. Unfortunately, it seems we can't use prepared statements for this, so we might need to construct the query at runtime and take special care to escape all the input prior to execution. See https://stackoverflow.com/questions/49156991/sqlite-range-bind-or-column-out-of-range-for-insert-statement

Reply:
Thinking about how to simplify these concerns, seems to me that the simplest approach would be:

- Update the DB with `TicketParam`s using an asynchronous process: possibly a goroutine for each O we're talking to
- I think in this case it's more valuable to prioritize sql injection protection over disk performance optimization, particularly because these operations are not on the critical path for video processing. This operation seems to be storing external inputs directly into the DB, which poses a real injection risk. I think it's best to use prepared statements in this case, and it's fine to write them one by one as they come in from the async goroutines.

wdyt?

Reply:
Another question I had - where are we capturing the need to augment this caching process with `TicketParam`s?

Reply:
Good to starting thinking about `GetOrchestratorInfo` retrieval. For when this happens in the background, it seems like it'd be fine to do async updating. However, when we're setting the cache for the first time on [startup](https://github.com/livepeer/go-livepeer/pull/642/files#diff-205b550ef0ac99b368da16cfb883960bR376), if the number of orchs is very large, wouldn't we experience a significant lag? Could we theoretically have two approaches?

Great idea on multiple INSERTs into a single statement @j0sh. I see the limit is 500 for that, so we'd be able to for now insert all at once. What do you mean by "escape all the input" @j0sh? SQL injections is a valid concern

Reply:
While this would likely be a "low-priority" background task, SQLite writes incur a global lock which blocks all other operations such as reads. We don't want to stall elsewhere in the critical path because of a blocked DB read, so we really should minimize writes as much as possible.